calculator_test.go

```diff
@@ -2,23 +2,172 @@ package calculator_test
 
 import (
 	"calculator"
+	"math/rand"
 	"testing"
+	"time"
 )
 
+const testcount = 100
+const testcaseMaxsize = 10
+
+type testCases struct {
+	name string
+	nums []float64
+	want float64
+}
+
+func TestAddRand(t *testing.T) {
+	t.Parallel()
+	seed := rand.NewSource(time.Now().UnixNano())
+	r := rand.New(seed)
+	count := 0
+	// run the test addcount times
+	for count < testcount {
+		// get a random size for the number of values
+		// to add together
+		nums := make([]float64, r.Intn(testcaseMaxsize))
+		// initialize nums slice to contain random values
+		var want float64 = 0
+		for i := range nums {
+			nums[i] = r.NormFloat64()
+			want += nums[i]
+		}
+
+		tc := testCases{name: "Random value data", nums: nums, want: want}
+		// call Add to get the sum of nums
+		got := calculator.Add(tc.nums...)
+		// did we get what we expected?
```

Comment:
Actually, your code is so clear it needs no explanatory comment! `if want != got` is absolutely unambiguous and even someone who doesn't know Go would probably understand this right away.

Reply:
Removed the more obvious comments.